client/app/components/DynamicComponent.jsx

```diff
@@ -0,0 +1,50 @@
+import { isFunction, isString } from 'lodash';
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const componentsRegistry = new Map();
+const activeInstances = new Set();
+
+export function registerComponent(alias, component) {
+  if (isString(alias) && (alias !== '')) {
+    componentsRegistry[alias] = isFunction(component) ? component : null;
+    // Refresh active DynamicComponent instances which use this alias
+    activeInstances.forEach((dynamicComponent) => {
+      if (dynamicComponent.props.is === alias) {
+        dynamicComponent.forceUpdate();
+      }
+    });
+  }
+}
+
+export function unregisterComponent(alias) {
+  registerComponent(alias, null);
+}
+
+export default class DynamicComponent extends React.Component {
+  static propTypes = {
+    is: PropTypes.string.isRequired,
```

Comment:
What about `alias` (the same as in `registerComponent`/`unregisterComponent`?

Reply:
The way I see it this property defines the name of the extension you're providing. Hence the idea of `name`/`key`.

It's not an alias, because it doesn't have another name. `use` is almost there, but it's not like DynamicComponent has any purpose of its own beside being a placeholder.

Reply:
okay, then `name` (because `key` is reserved by React)